src/main/java/com/elixr/poc/rest/response/PurchaseExceptionHandler.java

```diff
@@ -0,0 +1,33 @@
+package com.elixr.poc.rest.response;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ControllerAdvice
+public class PurchaseExceptionHandler {
+    private PurchaseErrorResponse errorResponse;
+
+    /*Handling the Exception and sending error message*/
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<?> myMessage(MethodArgumentNotValidException methodArgumentNotValidException) {
+        return getResponseEntity(methodArgumentNotValidException);
```

Comment:
This private method call is not needed here. you can directly keep the code inside this method itself

Reply:
Updated.